### ISSUE TITLE
automatic LLNODE_RANGESFILE

### ISSUE THREAD

**Issue by brendangregg** [nodejs/llnode#91]
Currently we do this for <tt>v8 findjsobjects</tt> to work:

```
# ./scripts/readelf2segments.py /var/cores/core.node.30833.1468367170 > core.30833.ranges
# export LLNODE_RANGESFILE=core.30833.ranges
```

It might be nice if llnode could do this automatically, even if it means llnode shelling out to call readelf and scraping the output (if that's possible in lldb).

**Comment by indutny** [comment]
This is not required with (I believe) lldb-3.9, as it can infer this information automatically with LLDB's APIs.

**Comment by indutny** [comment]
cc @hhellyer

**Comment by bnoordhuis** [comment]
Correct, llnode uses `lldb::SBProcess::GetMemoryRegions()` when available.

**Comment by hhellyer** [comment]
I added the memory region API to lldb a while ago: https://reviews.llvm.org/D20565, I think it was released in lldb 3.9.
You need to build against at least a 3.9 level of lldb with the 3.9 headers. Installing via the npm installer on Linux should get that right.

On Mac it's more complicated as Apple change the version numbers so it's hard to map the version numbers of lldb across but I have confirmed the latest lldb in Xcode has that API and updated the installation script: https://github.com/nodejs/llnode/pull/79

You can tell if it's enabled and you've built against the right headers as the #ifdef at: https://github.com/nodejs/llnode/blob/master/src/llnode.cc#L316 ensures the help text explaining to use the LLNODE_RANGESFILE env var is only printed if you need to set it.

**Comment by brendangregg** [comment]
Wow, great, thanks! I've updated my (makeshift) [docs](http://www.brendangregg.com/blog/2016-07-13/llnode-nodejs-memory-leak-analysis.html), and will close this ticket.
